models/user/user_list.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2025 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package user
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+)
+
+func GetUsersMapByIDs(ctx context.Context, userIDs []int64) (map[int64]*User, error) {
+	userMaps := make(map[int64]*User, len(userIDs))
+	left := len(userIDs)
+	for left > 0 {
+		limit := db.DefaultMaxInSize
+		if left < limit {
+			limit = left
+		}
+		err := db.GetEngine(ctx).
+			In("id", userIDs[:limit]).
+			Find(&userMaps)
+		if err != nil {
+			return nil, err
+		}
+		left -= limit
+		userIDs = userIDs[limit:]
+	}
+	return userMaps, nil
+}
+
+func MustGetUserFromMap(userID int64, usererMaps map[int64]*User) *User {
+	if userID == ActionsUserID {
+		return NewActionsUser()
+	}
+	if userID <= 0 {
+		return nil
+	}
+	user, ok := usererMaps[userID]
+	if !ok {
+		return NewGhostUser()
+	}
```

Comment:
Although it is old code, I think it should match `GetPossibleUserByID`: always use ghost user for `-1`, but not return early.